snowflake-table-grant-all/variables.tf

```diff
@@ -49,3 +49,14 @@ variable with_grant_option {
 
   default = null
 }
+
+variable per_privilege_grants {
+  type        = map(object({ roles : list(string), shares : list(string) }))
+  description = <<EOF
+  A map of privileges to authorized roles and shares. Privileges must be UPPER case.
```

Comment:
To prevent errors when the privs are in lower case, is there a way for us to convert privs to uppercase through the provider?

Reply:
why not convert them to upper in terraform?

Reply:
I think the provider is already case-insensitive. The issue here is more with looking up a key in this map being case-sensitive.

Reply:
Do `make plan`s fail when the privs are lowercase? Or you only see that you did the privs wrong in the apply?

Reply:
they'd be ignored

Reply:
>why not convert them to upper in terraform?
I could do something like that:
`upper_per_privilege_grants = {for priv, grant in var.per_privilege_grants: upper(priv) => grant }`
But don't think it would be well-defined for all inputs.